src/core/jsproxy.c

```diff
@@ -136,6 +136,20 @@ JsProxy_typeof(PyObject* self, void* _unused)
   return result;
 }
 
+static PyObject*
+JsProxy_js_id(PyObject* self, void* _unused)
+{
+  PyObject* result = NULL;
+
+  JsRef idval = JsProxy_REF(self);
+  int x[2] = { (int)Py_TYPE(self), (int)idval };
```

Comment:
So we add the type and hash because python hash is an identity for integers otherwise, as you said offline. But why it's a problem actually? Is there a situation where an object has the same idval but different types? 

In any case it would be good to add an inline comment explaining why this is done.

Reply:
Well imagine you have the following code:

```py
def store_js_object(x):
	if instanceof(x, JsProxy):
		my_set.add(x.js_id)
	else:
		my_set.add(x)
```
so some types get converted others get proxied. This is perfectly reasonable code. However, it wouldn't work very well without this becuase you would have `JsProxies` with `js_id` things like 5, 7, 11, 17 but you might also have these numbers appearing. If we ensure that `js_id` is a uniformly distributed 32 bit integer, it's much less likely for there to be a collision.

Reply:
Though it would be convenient for debugging internals to add a separate `_js_id` function to `_pyodide_core` that takes a `JsProxy` as argument and returns the hiwire key.

Reply:
Added `hiwire_id` as a private function to `_pyodide_core` to return the hiwire id of the `JsProxy`.